group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/RangeAssignorTest.java

```diff
@@ -0,0 +1,574 @@
+
+    @Test
+    public void testReassignmentNumConsumersGreaterThanNumPartitionsWhenOneConsumerAdded() {
+        Map<Uuid, AssignmentTopicMetadata> topics = new HashMap<>();
+        topics.put(topic1Uuid, new AssignmentTopicMetadata(2));
+        topics.put(topic2Uuid, new AssignmentTopicMetadata(2));
+
+        Map<String, AssignmentMemberSpec> members = new HashMap<>();
+        // Initial Subscriptions: A -> T1, T2 | B -> T1, T2 | C -> T1, T2
+
+        Map<Uuid, Set<Integer>> currentAssignmentForA = new HashMap<>();
+        currentAssignmentForA.put(topic1Uuid, Collections.singleton(0));
+        currentAssignmentForA.put(topic2Uuid, Collections.singleton(0));
+        members.put(consumerA, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic2Uuid),
+            currentAssignmentForA)
+        );
+
+        Map<Uuid, Set<Integer>> currentAssignmentForB = new HashMap<>();
+        currentAssignmentForB.put(topic1Uuid, Collections.singleton(1));
+        currentAssignmentForB.put(topic2Uuid, Collections.singleton(1));
+        members.put(consumerB, new AssignmentMemberSpec(
+            Optional.empty(),
+            Optional.empty(),
+            Arrays.asList(topic1Uuid, topic2Uuid),
+            currentAssignmentForB)
+        );
+
+        // Add a new consumer to trigger a re-assignment
+        members.put(consumerC, new AssignmentMemberSpec(
+                Optional.empty(),
+                Optional.empty(),
+                Arrays.asList(topic1Uuid, topic2Uuid),
+                Collections.emptyMap())
+        );
+
+        AssignmentSpec assignmentSpec = new AssignmentSpec(members, topics);
+        GroupAssignment computedAssignment = assignor.assign(assignmentSpec);
+
+        Map<Uuid, Set<Set<Integer>>> expectedAssignment = new HashMap<>();
+        // Topic 1 Partitions Assignment
+        mkAssignment(expectedAssignment, topic1Uuid, Collections.singleton(0));
+        mkAssignment(expectedAssignment, topic1Uuid, Collections.singleton(1));
+        // Topic 2 Partitions Assignment
+        mkAssignment(expectedAssignment, topic2Uuid, Collections.singleton(0));
+        mkAssignment(expectedAssignment, topic2Uuid, Collections.singleton(1));
+
+        // Test for stickiness
+        assertEquals(computedAssignment.members().get(consumerA)
+            .targetPartitions(), new HashMap<>(currentAssignmentForA),
+            "Stickiness test failed for consumer A");
+        assertEquals(computedAssignment.members().get(consumerB)
+            .targetPartitions(), new HashMap<>(currentAssignmentForB),
+            "Stickiness test failed for consumer B");
+
+        // Consumer C shouldn't get any assignment, due to stickiness A, B retain their assignments
+        assertNull(computedAssignment.members().get(consumerC));
+        assertAssignment(expectedAssignment, computedAssignment);
+        assertCoPartitionJoinProperty(computedAssignment);
```

Comment:
In this case, the expected assignment seems to be deterministic so we could just use `assertEquals`. This seems to be true for most of the `testReassignment` test cases.

Reply:
all tests are checked with 1:1 mapping now so this is taken care of now